File: 50-PowxN/50-PowxN.py
# Last updated: 6/11/2026, 11:37:49 AM
class Solution:
    def myPow(self, x: float, n: int) -> float:
        def fun(x,n):
            if(n==0):
                return 1
            elif(n==1):
                return x
            else:
                if(n%2 == 0):
                    return fun(x*x,n//2)
                else:
                    return x*fun(x,n-1)
        if(n<0):
            x = 1/x
            n = -n
            return fun(x,n)
        else:
            return fun(x,n)

File: 50-PowxN/50-PowxN.cpp
// Last updated: 6/11/2026, 11:37:44 AM
class Solution {
public:
    double myPow(double x, long long n) {
        if(n<0){
            return 1.0/myPow(x, abs(n));
        }
        if(n == 0) return 1;
        if(x == 1) return x;
        double ans = myPow(x, n/2);
        ans = ans*ans;
        if(n%2 == 1){
            ans = ans*x;
        }
        return ans;
    }
};